flang-rt/lib/runtime/pointer.cpp

```diff
@@ -267,8 +267,8 @@ bool RTDEF(PointerIsAssociatedWith)(
   if (!target) {
     return pointer.raw().base_addr != nullptr;
   }
-  if (!target->raw().base_addr || target->ElementBytes() == 0
-      || target->Elements() == 0) {
```

Comment:
Is this invalid C++ code that was excepted by the compiler? It is weird to me that the compiler didn't reject this or treat it the same as the changed code.

Reply:
Never mind, I see what happened here.